service/controller/clusterapi/resource.go

```diff
@@ -352,7 +353,10 @@ func New(config Config) ([]resource.Interface, error) {
 	}
 
 	var alertmanagerWiringResource resource.Interface
-	{
+	// This resource creates a static secret to connect Prometheus to Alertmanager. When using mimir, this is not needed anymore
+	if config.MimirEnabled {
+		alertmanagerWiringResource = &noop.Resource{}
```

Comment:
Adding a noop operation to not create useless secret in mimir backed installations